Wait for deployment to complete before displaying output

vivsriaus opened the pull request:
Wait for deployment to complete before displaying output

azurecla commented:
Hi __@vivsriaus__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (visriniv). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

amarzavery commented on lib/commands/arm/group/group.deployment._js:
```diff
@@ -50,8 +50,27 @@ exports.init = function (cli) {
+      var subscription = profile.current.getSubscription(options.subscription);
+      var client = utils.createResourceClient(subscription);      
+      
+      var d;
```
please provide meaningful variable names

vivsriaus replied:
Fixed

amarzavery commented on lib/commands/arm/group/group.deployment._js:
```diff
@@ -50,8 +50,27 @@ exports.init = function (cli) {
+        function (log, _) {          
+          do {
+            d = client.deployments.get(resourceGroup, deployment.name, _).deployment;            
+          } while (d.properties.provisioningState == 'Running' || d.properties.provisioningState == 'Accepted');         
```
in javascript, we use triple equal to for comparison. Read [this](http://stackoverflow.com/questions/359494/does-it-matter-which-equals-operator-vs-i-use-in-javascript-comparisons)

vivsriaus replied:
Ah, thanks for the info.

amarzavery commented on lib/commands/arm/group/group.deployment._js:
```diff
@@ -50,8 +50,27 @@ exports.init = function (cli) {
+            d = client.deployments.get(resourceGroup, deployment.name, _).deployment;            
+          } while (d.properties.provisioningState == 'Running' || d.properties.provisioningState == 'Accepted');         
+
+          if (d.properties.provisioningState != 'Succeeded') {
```
not double equal to (!==)

vivsriaus replied:
Fixed

amarzavery commented on lib/commands/arm/group/group.deployment._js:
```diff
@@ -50,8 +50,27 @@ exports.init = function (cli) {
+          } while (d.properties.provisioningState == 'Running' || d.properties.provisioningState == 'Accepted');         
+
+          if (d.properties.provisioningState != 'Succeeded') {
+            console.log('Deployment provisioning state was not successful. State is %s', d.properties.provisioningState);            
```
log.error(). Never, use console.log()

vivsriaus replied:
Sounds good. thanks

amarzavery commented on lib/commands/arm/group/group.deployment._js:
```diff
@@ -50,8 +50,27 @@ exports.init = function (cli) {
+          }
+
+          if (d.properties.provisioningState == 'Succeeded') {
+            log.info('Deployment succeeded');
```
this information is provided any which ways as a part of displayDeployment() function. So you can remove the if check for succeeded.

vivsriaus replied:
Makes sense

amarzavery commented:
the group log show test fails because of this change.

amarzavery commented:
why would you make the command poll for succeeded state> That will make it time consuming. Indirectly tests will take more time as well. 

just trying to understand the reason behind this change.

vivsriaus commented:
I fixed the test by capturing new records. Updated the record logs with my latest commit.

With respect to the change, the group deployment create command originally returned as soon as the create command was issued (where users would see a provisioningState of Accepted as soon as they hit enter). This is not the ideal, since users would not know if the deployment failed or not. We're trying to mirror the experience in powershell, where we wait for the deployment to succeed/fail and then call the displayDeployment(), instead of calling displayDeployment() as soon as the /create command as issued.
